### ISSUE TITLE
[no-unused-variable] Unused destructured variables ignored with object rest syntax.

### ISSUE THREAD

**Issue by phyllisstein** [palantir/tslint#3800]
### Bug Report

- __TSLint version__: `v5.9.1`
- __TypeScript version__: `v2.8.1`
- __Running TSLint via__: (pick one) Atom's `linter-tslint`

#### TypeScript code being linted

```ts
function restInPeace() {
  const anObject = { a: 'a', b: 'b' }
  const {
    a,
    b,
    ...obj
  } = anObject
}
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "warning",
  "rules": {
    "no-unused-variable": [
      true,
      "check-parameters",
      { "ignore-pattern": "^_" }
    ]
  }
}
```

#### Actual behavior
TSLint flags `obj` as unused, and ignores `a` and `b`.

#### Expected behavior
It would be great if it were possible to at least configure this to catch `a` and `b`, in the style of ESLint, which offers an `ignoreRestSiblings` flag for [its unused variable check](https://eslint.org/docs/4.0.0/rules/no-unused-vars). My personal preference is to check rest siblings when destructuring, because it often alerts me to cases when I meant to pass a React prop down to a child component and forgot.

**Comment by johnwiseheart** [comment]
I think it makes sense to add an extra flag for this.

**Comment by ajafff** [comment]
Turns out the custom rule [`no-unused`](https://github.com/ajafff/tslint-consistent-codestyle/blob/master/docs/no-unused.md) from my package `tslint-consistent-codestyle` already checks siblings of object rest. You need to explicitly ignore the variable by adding an underscore prefix.

**Comment by JoshuaKGoldberg** [comment]
Awesome - closing as external on both accounts.
